keps/104-StartupPolicy/README.md

```diff
@@ -0,0 +1,428 @@
+
+The API below would fit my purpose.
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: driver-ready-worker-start
+spec:
+  startupPolicy:
+     startupPolicyOrder: InOrder
+  replicatedJobs:
+  - name: driver
+    replicas: 1
+    template:
+      spec:
+        # Set backoff limit to 0 so job will immediately fail if any pod fails.
+        backoffLimit: 0 
+        completions: 1
+        parallelism: 1
+        template:
+          spec:
+            containers:
+            - name: driver
+              image: bash:latest
+              command:
+              - bash
+              - -xc
+              - |
+                sleep 10000
+  - name: workers
+    replicas: 1
+    template:
+      spec:
+        backoffLimit: 0 
+        completions: 2
+        parallelism: 2
+        template:
+          spec:
+            containers:
+            - name: worker
+              image: bash:latest
+              command:
+              - bash
+              - -xc
+              - |
+                sleep 10
+```
+
+#### Story 2
+
+As a user, I want my replicated jobs to start in order.
+
+One area that this could be useful is if someone has a message queue replicatedJob (e.g., Redis), followed by a driver replicatedJob, followed by a worker replicated job.
+
+In this case, I want message queue to start first, then the driver, and then finally the worker.  
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: messagequeue-driver-worker
+spec:
+  startupPolicy:
+     startupPolicyOrder: InOrder
+  replicatedJobs:
+  - name: messagequeue
+    replicas: 1
+    template:
+      spec:
+        # Set backoff limit to 0 so job will immediately fail if any pod fails.
+        backoffLimit: 0 
+        completions: 1
+        parallelism: 1
+        template:
+          spec:
+            containers:
+            - name: messagequeue
+              image: bash:latest
+              command:
+              - bash
+              - -xc
+              - |
+                sleep 100
+  - name: driver
+    replicas: 2
+    template:
+      spec:
+        backoffLimit: 0 
+        completions: 2
+        parallelism: 2
+        template:
+          spec:
+            containers:
+            - name: driver
+              image: bash:latest
+              command:
+              - bash
+              - -xc
+              - |
+                sleep 10
+  - name: worker
+    replicas: 2
+    template:
+      spec:
+        backoffLimit: 0 
+        completions: 2
+        parallelism: 2
+        template:
+          spec:
+            containers:
+            - name: worker
+              image: bash:latest
+              command:
+              - bash
+              - -xc
+              - |
+                sleep 10
+```
+
+### Risks and Mitigations
+
+To avoid the JobSet become a half-baked workflow manager, we will not allow for starting particular replicated jobs.
```

Comment:
```suggestion
To avoid the JobSet become a half-baked workflow manager, the API will not allow describing DAGs. The goal is to focus on supporting startup sequence that HPC and ML training jobs typically require.
```

Reply:
updated.